app.py

```diff
@@ -92,289 +82,212 @@
+        return {"user_id": user.get_id()}
 
     except Exception as e:
         logger.error(f"Error occurred: {e}")
         logger.error("Traceback: " + traceback.format_exc())
-        return jsonify(
-            error="Failed to create user!", message=str(e)
-        ), HTTPStatus.INTERNAL_SERVER_ERROR
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Failed to create user!")
 
 
-@auth.verify_password
-def verify_password(username, password):
+async def verify_password(username: str, password: str):
     if not username:
-        return jsonify(
-            error="Missing email address!",
-            message="The request is missing the 'email' parameter. Please provide a valid email address to proceed.",
-        ), HTTPStatus.BAD_REQUEST
+        raise HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail="Missing email address!")
 
     try:
-        with connection_manager as manager:
-            with manager.get_cursor() as cursor:
-                # Check if the user exists in the database
+        async with connection_manager as manager:
+            async with manager.get_cursor() as cursor:
                 try:
-                    user = asyncio.run(get_user(cursor, username))
+                    user = await get_user(cursor, username)
                 except Exception as e:
                     logger.error(f"Error occurred: {e}")
                     logger.error("Traceback: " + traceback.format_exc())
-                    return jsonify(
-                        error="Authentication error!",
-                        message=str(e),
-                    ), HTTPStatus.UNAUTHORIZED
+                    raise HTTPException(status_code=HTTPStatus.UNAUTHORIZED, detail="Authentication error!")
 
         if user is None:
-            return jsonify(
-                error="Unknown user!",
-                message="The email provided does not match with any known user.",
-            ), HTTPStatus.UNAUTHORIZED
+            raise HTTPException(status_code=HTTPStatus.UNAUTHORIZED, detail="Unknown user!")
 
-        return jsonify(user_id=user.get_id()), HTTPStatus.OK
+        return {"user_id": user.get_id()}
 
     except Exception as err:
         logger.error(f"Error occurred: {err}")
         logger.error("Traceback: " + traceback.format_exc())
-        return jsonify(
-            error="Internal server error!", message=str(err)
-        ), HTTPStatus.INTERNAL_SERVER_ERROR
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal server error!")
 
 
-@app.route("/inspections", methods=["POST"])
-@auth.login_required
-@cross_origin(origins=FRONTEND_URL)
-@swag_from("docs/swagger/create_inspection.yaml")
-def create_inspection():  # pragma: no cover
+@app.post("/inspections")
+async def create_inspection(form: dict, files: list[UploadFile] = File(...), credentials: HTTPBasicCredentials = Depends(auth)):
```

Comment:
Try using a pydantic model of the expected form instead of dict. FastAPI and pydantic go hand in hand.

Reply:
Also specify a return type as a pydantic model and see the magic happen.